pkg/azureconditions/deploymenthandler.go

```diff
@@ -20,16 +22,47 @@ const (
 	DeploymentProvisioningStateFailed        = "Failed"
 )
 
-func (r *Resource) checkIfDeploymentIsSuccessful(ctx context.Context, deploymentsClient *resources.DeploymentsClient, cr conditions.CR, deploymentName string, conditionType capi.ConditionType) (bool, error) {
+type DeploymentCheckerConfig struct {
+	CtrlClient client.Client
+	Logger     micrologger.Logger
+}
+
+// Resource ensures that AzureMachinePool Status Conditions are set.
+type DeploymentChecker struct {
```

Comment:
Moved this to a shared sub-package, so it can be reused in every conditions handler where some Azure deployment is checked. This will later be moved to a separate `giantswarm/azure-conditions-handler` module.

Reply:
Yeah, I missed that :+1: